### PR TITLE
Timestamp from str to transcription

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: app/timecode/views.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        w_count = 0
+        while w_count < w_len:
+            resp.append((j1 + w_count, round((t_offset + w_count * coef) / 1000, 1)))
+            w_count += 1
+        return resp
+    except Exception as e:
+        current_app.logger.error(e)
+
+
+# must be synchronized with client side clean text cause we use letter position for find timestamp
```
Comment (bbfsdev) [inline app/timecode/views.py:208]: Add exact path the the relevant file and function name. Link to github or something.

File: app/timecode/views.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+def clear_str(text):
+    pattern = re.compile(r'[".,\/#!$%\^&\*;:{}=\-_`~()\[\]‘’”“]')
```
Comment (bbfsdev) [inline app/timecode/views.py]: You can compile this once and not each time you call this function

File: app/timecode/views.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+from pprint import pprint
+
+from flask import Blueprint, current_app, request
+from flask.helpers import make_response
+import webvtt
+
+from app.doc2html.conversionFunctions import docx_to_text
+from app.doc2html.views import process_docx_uid
+
+blueprint = Blueprint('time_code', __name__)
```
Comment (bbfsdev) [inline app/timecode/views.py:14]: What is a blueprint?

File: app/timecode/views.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+FIX_DERIVE_ZERO = 10 ** -5
+
+vtt_q = """
```
Comment (bbfsdev) [inline app/timecode/views.py:16]: Please add comment I don't know what this variable is for.

File: app/timecode/views.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+file_q = """
+        select f.* from files f
+        INNER JOIN content_units cu ON cu.id = f.content_unit_id
+        WHERE cu.uid = '%s'
+        AND f.language = '%s'
+        """
+
+
+@blueprint.route('/time_code')
+def timecode():
```
Comment (bbfsdev) [inline app/timecode/views.py:32]: Please add comments and descriptions to the whole file.

**Review by bbfsdev — CHANGES_REQUESTED** [review]
Tooltip is wrong:
image

Please describe the algorithm you are using with comments inside the code.

File: app/timecode/views.py (hunk 2 above)
Comment (davgur) [reply app/timecode/views.py]: fix

File: app/timecode/views.py (hunk 1 above)
Comment (davgur) [reply app/timecode/views.py:208]: added

File: app/timecode/views.py (hunk 3 above)
Comment (davgur) [reply app/timecode/views.py:14]: its main python library for build rest

File: app/timecode/views.py (hunk 4 above)
Comment (davgur) [reply app/timecode/views.py:16]: removed

File: app/timecode/views.py (hunk 5 above)
Comment (davgur) [reply app/timecode/views.py:32]: added

**Pull request merged**